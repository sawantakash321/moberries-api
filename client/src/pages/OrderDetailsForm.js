import Grid from '@material-ui/core/Grid';
import MenuItem from '@material-ui/core/MenuItem';
import TextField from '@material-ui/core/TextField';
import Typography from '@material-ui/core/Typography';
import PropTypes from 'prop-types';
import React from 'react';

const duration = [
  {
    value: '3',
    label: '3',
  },
  {
    value: '6',
    label: '6',
  },
  {
    value: '12',
    label: '12',
  },
];

const gigabytes = [
  {
    value: '3',
    label: '3',
  },
  {
    value: '5',
    label: '5',
  },
  {
    value: '10',
    label: '10',
  },
  {
    value: '20',
    label: '20',
  },
  {
    value: '30',
    label: '30',
  },
  {
    value: '50',
    label: '50',
  },
];

const upfrontPayment = [
  {
    value: 'true',
    label: 'Yes',
  },
  {
    value: 'false',
    label: 'No',
  },
];

const OrderDetailsForm = props => {
  const {
    values,
    handleChange,
    errors: {errors},
  } = props;

  return (
    <React.Fragment>
      <Typography variant="h6" gutterBottom>
        Select subscription parameters:
      </Typography>
      <br />
      <Grid container spacing={24} alignItems="center" direction="column" justify="center">
        <Grid item xs={12}>
          <TextField
            id="duration"
            select
            label="Duration"
            value={values.duration}
            onChange={handleChange('duration')}
            name="duration"
            margin="normal"
            variant="outlined"
            error={values.duration === ''}
            helperText={'Please select the duration in Months' + (errors.isValid === false ? errors.duration : ' ')}>
            {duration.map(option => (
              <MenuItem key={option.value} value={option.value}>
                {option.label}
              </MenuItem>
            ))}
          </TextField>
          <br />
        </Grid>
        <Grid item xs={12}>
          <TextField
            id="gigabytes"
            select
            label="Gigabytes"
            name="gigabytes"
            value={values.gigabytes}
            onChange={handleChange('gigabytes')}
            margin="normal"
            variant="outlined"
            error={values.gigabytes === ''}
            helperText={
              `Please select the amount of gigabytes for ${values.duration} Months` +
              (errors.isValid === false ? errors.gigabytes : ' ')
            }>
            {gigabytes.map(option => (
              <MenuItem key={option.value} value={option.value}>
                {option.label}
              </MenuItem>
            ))}
          </TextField>
          <br />
        </Grid>

        <Grid item xs={12}>
          <TextField
            id="upfrontPayment"
            select
            label="Upfront Payment"
            name="upfrontPayment"
            value={values.upfrontPayment}
            onChange={handleChange('upfrontPayment')}
            margin="normal"
            variant="outlined"
            error={values.upfrontPayment === ''}
            helperText={'Would you like to pay Upfront?' + (errors.isValid === false ? errors.upfrontPayment : ' ')}>
            {upfrontPayment.map(option => (
              <MenuItem key={option.value} value={option.value}>
                {option.label}
              </MenuItem>
            ))}
          </TextField>
          <br />
        </Grid>
        <br />
        <Typography variant="h6" gutterBottom>
          Total price: {`$${values.computedPrice}`}
        </Typography>
      </Grid>
    </React.Fragment>
  );
};

OrderDetailsForm.propTypes = {
  values: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired,
  handleChange: PropTypes.func.isRequired,
};

export default OrderDetailsForm;
