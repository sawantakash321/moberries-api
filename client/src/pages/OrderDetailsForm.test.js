'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import OrderDetailsForm from './OrderDetailsForm';

// Snapshot test
it('matches the snapshot when there are no validation errors', () => {
  const values = {
    duration: '6',
    gigabytes: '20',
    upfrontPayment: 'true',
  };
  const errors = {
    errors: {
      duration: '',
      gigabytes: '',
      upfrontPayment: '',
    },
  };
  const handleChange = jest.fn();
  const tree = renderer
    .create(<OrderDetailsForm values={values} errors={{errors}} handleChange={handleChange} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
