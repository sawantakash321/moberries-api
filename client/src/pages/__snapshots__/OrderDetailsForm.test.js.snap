// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot when there are no validation errors 1`] = `
Array [
  <h6
    className="MuiTypography-root-1 MuiTypography-h6-18 MuiTypography-gutterBottom-28"
  >
    Select subscription parameters:
  </h6>,
  <br />,
  <div
    className="MuiGrid-container-37 MuiGrid-spacing-xs-24-61 MuiGrid-direction-xs-column-40 MuiGrid-align-items-xs-center-45 MuiGrid-justify-xs-center-54"
  >
    <div
      className="MuiGrid-item-38 MuiGrid-grid-xs-12-77"
    >
      <div
        className="MuiFormControl-root-134 MuiFormControl-marginNormal-135"
      >
        <label
          className="MuiFormLabel-root-149 MuiFormLabel-filled-153 MuiInputLabel-root-138 MuiInputLabel-formControl-143 MuiInputLabel-animated-146 MuiInputLabel-shrink-145 MuiInputLabel-outlined-148"
          data-shrink={true}
          htmlFor="duration"
        >
          Duration
        </label>
        <div
          className="MuiInputBase-root-176 MuiOutlinedInput-root-163 MuiInputBase-formControl-177"
          onClick={[Function]}
        >
          <fieldset
            aria-hidden={true}
            className="MuiPrivateNotchedOutline-root-193 MuiOutlinedInput-notchedOutline-170"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="MuiPrivateNotchedOutline-legend-194"
              style={
                Object {
                  "width": 0,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
          <div
            className="MuiSelect-root-156"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              className="MuiSelect-select-157 MuiSelect-selectMenu-160 MuiSelect-outlined-159 MuiInputBase-input-186 MuiOutlinedInput-input-171"
              id="select-duration"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              6
            </div>
            <input
              aria-describedby="duration-helper-text"
              id="duration"
              name="duration"
              type="hidden"
              value="6"
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-195 MuiSelect-icon-162"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
        <p
          className="MuiFormHelperText-root-208 MuiFormHelperText-contained-212 MuiFormHelperText-filled-214"
          id="duration-helper-text"
        >
          Please select the duration in Months 
        </p>
      </div>
      <br />
    </div>
    <div
      className="MuiGrid-item-38 MuiGrid-grid-xs-12-77"
    >
      <div
        className="MuiFormControl-root-134 MuiFormControl-marginNormal-135"
      >
        <label
          className="MuiFormLabel-root-149 MuiFormLabel-filled-153 MuiInputLabel-root-138 MuiInputLabel-formControl-143 MuiInputLabel-animated-146 MuiInputLabel-shrink-145 MuiInputLabel-outlined-148"
          data-shrink={true}
          htmlFor="gigabytes"
        >
          Gigabytes
        </label>
        <div
          className="MuiInputBase-root-176 MuiOutlinedInput-root-163 MuiInputBase-formControl-177"
          onClick={[Function]}
        >
          <fieldset
            aria-hidden={true}
            className="MuiPrivateNotchedOutline-root-193 MuiOutlinedInput-notchedOutline-170"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="MuiPrivateNotchedOutline-legend-194"
              style={
                Object {
                  "width": 0,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
          <div
            className="MuiSelect-root-156"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              className="MuiSelect-select-157 MuiSelect-selectMenu-160 MuiSelect-outlined-159 MuiInputBase-input-186 MuiOutlinedInput-input-171"
              id="select-gigabytes"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              20
            </div>
            <input
              aria-describedby="gigabytes-helper-text"
              id="gigabytes"
              name="gigabytes"
              type="hidden"
              value="20"
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-195 MuiSelect-icon-162"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
        <p
          className="MuiFormHelperText-root-208 MuiFormHelperText-contained-212 MuiFormHelperText-filled-214"
          id="gigabytes-helper-text"
        >
          Please select the amount of gigabytes for 6 Months 
        </p>
      </div>
      <br />
    </div>
    <div
      className="MuiGrid-item-38 MuiGrid-grid-xs-12-77"
    >
      <div
        className="MuiFormControl-root-134 MuiFormControl-marginNormal-135"
      >
        <label
          className="MuiFormLabel-root-149 MuiFormLabel-filled-153 MuiInputLabel-root-138 MuiInputLabel-formControl-143 MuiInputLabel-animated-146 MuiInputLabel-shrink-145 MuiInputLabel-outlined-148"
          data-shrink={true}
          htmlFor="upfrontPayment"
        >
          Upfront Payment
        </label>
        <div
          className="MuiInputBase-root-176 MuiOutlinedInput-root-163 MuiInputBase-formControl-177"
          onClick={[Function]}
        >
          <fieldset
            aria-hidden={true}
            className="MuiPrivateNotchedOutline-root-193 MuiOutlinedInput-notchedOutline-170"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="MuiPrivateNotchedOutline-legend-194"
              style={
                Object {
                  "width": 0,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
          <div
            className="MuiSelect-root-156"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              className="MuiSelect-select-157 MuiSelect-selectMenu-160 MuiSelect-outlined-159 MuiInputBase-input-186 MuiOutlinedInput-input-171"
              id="select-upfrontPayment"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              Yes
            </div>
            <input
              aria-describedby="upfrontPayment-helper-text"
              id="upfrontPayment"
              name="upfrontPayment"
              type="hidden"
              value="true"
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-195 MuiSelect-icon-162"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
        <p
          className="MuiFormHelperText-root-208 MuiFormHelperText-contained-212 MuiFormHelperText-filled-214"
          id="upfrontPayment-helper-text"
        >
          Would you like to pay Upfront? 
        </p>
      </div>
      <br />
    </div>
    <br />
    <h6
      className="MuiTypography-root-1 MuiTypography-h6-18 MuiTypography-gutterBottom-28"
    >
      Total price: 
      $undefined
    </h6>
  </div>,
]
`;
